src/main/groovy/io/codearte/gradle/nexus/infra/SimplifiedHttpJsonRestClient.groovy

```diff
@@ -1,77 +1,84 @@
 package io.codearte.gradle.nexus.infra
 
+import groovy.json.JsonOutput
+import groovy.json.JsonSlurper
 import groovy.transform.CompileStatic
 import groovy.util.logging.Slf4j
-import groovyx.net.http.ContentType
-import groovyx.net.http.HttpResponseDecorator
-import groovyx.net.http.HttpResponseException
-import groovyx.net.http.RESTClient
+import java.util.concurrent.TimeUnit
+import javax.annotation.concurrent.ThreadSafe
+import okhttp3.Credentials
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.Response
+import org.jetbrains.annotations.NotNull
 
 /**
  * Specialized REST client to communicate with Nexus.
- *
- * Note: The same as RESTClient from HTTP Builder this class is not thread-safe.
  */
 @CompileStatic
 @Slf4j
+@ThreadSafe
 class SimplifiedHttpJsonRestClient {
 
-    private enum RequestType {
-        GET, POST
-    }
+    private final OkHttpClient restClient
+
+    SimplifiedHttpJsonRestClient(OkHttpClient restClient, String username, String password) {
+        OkHttpClient.Builder clientBuilder = restClient.newBuilder()
+            .connectTimeout(60, TimeUnit.SECONDS)
```

Comment:
I noticed some timeouts on the e2e test and wonder if it's because okhttp defaults timeouts to 10s, and from what I could understand apache is usually 60s. So I set to 60s here to be closer to what it was previously (I think).

Reply:
HTTP Builder most likely had longer default timeouts than OkHttp. Let's check it out how often those values will be not enough with Sonatype Nexus :).